<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="pandoc" />

        <meta name="author" content="Derek Michael Wright derek.wright@usask.ca" />
    
    
    <title>Focusing the GWAS Lens on days to flower using latent variable phenotypes derived from global multi-environment trials</title>

        <script src="README_files/header-attrs-2.23/header-attrs.js"></script>
        <script src="README_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link href="README_files/bootstrap-3.3.7/css/bootstrap.min.css" rel="stylesheet" />
        <script src="README_files/bootstrap-3.3.7/js/bootstrap.min.js"></script>
        <script src="README_files/navigation-1.1/tabsets.js"></script>
        <link href="README_files/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
        <script src="README_files/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
        <link href="README_files/readthedown-0.1/readthedown.css" rel="stylesheet" />
        <link href="README_files/readthedown-0.1/readthedown_fonts_embed.css" rel="stylesheet" />
        <script src="README_files/readthedown-0.1/readthedown.js"></script>
    
    
    
        <link rel="stylesheet" href="custom.css" type="text/css" />
    
    <!-- tabsets -->
    <script>
      $(document).ready(function () {
	  window.buildTabsets("toc");
      });
      $(document).ready(function () {
	  $('.tabset-dropdown > .nav-tabs > li').click(function () {
	      $(this).parent().toggleClass('nav-tabs-open')
	  });
      });
    </script>

    <!-- code folding -->
    
    <!-- code download -->
    
    <!-- tabsets dropdown -->

    <style type="text/css">
      .tabset-dropdown > .nav-tabs {
	  display: inline-table;
	  max-height: 500px;
	  min-height: 44px;
	  overflow-y: auto;
	  background: white;
	  border: 1px solid #ddd;
	  border-radius: 4px;
      }
      
      .tabset-dropdown > .nav-tabs > li.active:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
	  content: "&#xe258;";
	  border: none;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs > li.active {
	  display: block;
      }

      .tabset-dropdown > .nav-tabs > li.active a {
  	  padding: 0 15px !important;
      }

      .tabset-dropdown > .nav-tabs > li > a,
      .tabset-dropdown > .nav-tabs > li > a:focus,
      .tabset-dropdown > .nav-tabs > li > a:hover {
	  border: none;
	  display: inline-block;
	  border-radius: 4px;
	  background-color: transparent;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li {
	  display: block;
	  float: none;
      }
      
      .tabset-dropdown > .nav-tabs > li {
	  display: none;
	  margin-left: 0 !important;
      }
    </style>
    
</head>

<body class="preload">

   	
         <!-- readthedown start -->   
   <div id="content" data-toggle="wy-nav-shift">
     <nav id="nav-top" role="navigation" aria-label="top navigation">
       <a role="button" href="#" data-toggle="wy-nav-top"><span class="glyphicon glyphicon-menu-hamburger"></span></a>
     </nav>
         
   
        
      <h1 class="title">Focusing the GWAS <em>Lens</em> on days to
flower using latent variable phenotypes derived from global
multi-environment trials</h1>
         <h1 class="subtitle"><em>The Plant Genome</em>. (2022)
e20269.</h1>
   
         <!-- readthedown authors -->
   <div id="sidebar">
    <h2><a href="#content">Focusing the GWAS <em>Lens</em> on days to
flower using latent variable phenotypes derived from global
multi-environment trials</a></h2>
    <div id="toc">
      <ul>
      <li><a href="#contents" id="toc-contents">Contents</a></li>
      <li><a href="#agile-project" id="toc-agile-project">AGILE
      Project</a>
      <ul>
      <li><a href="#collaborators"
      id="toc-collaborators">Collaborators</a></li>
      <li><a href="#sponsors" id="toc-sponsors">Sponsors</a></li>
      </ul></li>
      <li><a href="#figures" id="toc-figures">Figures</a>
      <ul>
      <li><a href="#figure-1" id="toc-figure-1">Figure 1</a></li>
      <li><a href="#figure-2" id="toc-figure-2">Figure 2</a></li>
      <li><a href="#figure-3" id="toc-figure-3">Figure 3</a></li>
      <li><a href="#figure-4" id="toc-figure-4">Figure 4</a></li>
      <li><a href="#figure-5" id="toc-figure-5">Figure 5</a></li>
      </ul></li>
      <li><a href="#supplemental-tables"
      id="toc-supplemental-tables">Supplemental Tables</a>
      <ul>
      <li><a href="#supplemental-table-1"
      id="toc-supplemental-table-1">Supplemental Table 1</a></li>
      <li><a href="#supplemental-table-2"
      id="toc-supplemental-table-2">Supplemental Table 2</a></li>
      </ul></li>
      <li><a href="#supplemental-figures"
      id="toc-supplemental-figures">Supplemental Figures</a>
      <ul>
      <li><a href="#supplemental-figure-1"
      id="toc-supplemental-figure-1">Supplemental Figure 1</a></li>
      <li><a href="#supplemental-figure-2"
      id="toc-supplemental-figure-2">Supplemental Figure 2</a></li>
      <li><a href="#supplemental-figure-3"
      id="toc-supplemental-figure-3">Supplemental Figure 3</a></li>
      <li><a href="#supplemental-figure-4"
      id="toc-supplemental-figure-4">Supplemental Figure 4</a></li>
      <li><a href="#supplemental-figure-5"
      id="toc-supplemental-figure-5">Supplemental Figure 5</a></li>
      <li><a href="#supplemental-figure-6"
      id="toc-supplemental-figure-6">Supplemental Figure 6</a></li>
      <li><a href="#supplemental-figure-7"
      id="toc-supplemental-figure-7">Supplemental Figure 7</a></li>
      </ul></li>
      <li><a href="#additional-figures"
      id="toc-additional-figures">Additional Figures</a>
      <ul>
      <li><a href="#additional-figure-1"
      id="toc-additional-figure-1">Additional Figure 1</a></li>
      <li><a href="#additional-figure-2"
      id="toc-additional-figure-2">Additional Figure 2</a></li>
      <li><a href="#additional-figure-3"
      id="toc-additional-figure-3">Additional Figure 3</a></li>
      <li><a href="#additional-figure-4"
      id="toc-additional-figure-4">Additional Figure 4</a></li>
      <li><a href="#additional-figure-5"
      id="toc-additional-figure-5">Additional Figure 5</a></li>
      <li><a href="#grouped-manhattan-plots"
      id="toc-grouped-manhattan-plots">Grouped Manhattan Plots</a></li>
      <li><a href="#facetted-multi-modeled-manhattan-plots"
      id="toc-facetted-multi-modeled-manhattan-plots">Facetted &amp;
      Multi-Modeled Manhattan Plots</a></li>
      <li><a href="#marker-plots" id="toc-marker-plots">Marker
      Plots</a></li>
      <li><a href="#flowering-time-genes"
      id="toc-flowering-time-genes">Flowering Time Genes</a></li>
      </ul></li>
      </ul>
    </div>
    <div id="postamble" data-toggle="wy-nav-shift" class="status">
                  <p class="author"><span class="glyphicon glyphicon-user"></span> Derek
Michael Wright <a href="mailto:derek.wright@usask.ca"
class="email">derek.wright@usask.ca</a></p>
                        <p class="date"><span class="glyphicon glyphicon-calendar"></span> 25-10-2022</p>
          </div>
   </div>
     

   
      
   
<!-- Don't indent these lines or it will mess pre blocks indentation --> 
<div id="main">
<div style="page-break-after: always;"></div>
<hr />
<blockquote>
<p><a href="https://doi.org/10.1002/tpg2.20269">Sandesh Neupane, Derek
M. Wright, Raul O. Martinez, Jakob Butler, Jim L. Weller, Kirstin E.
Bett. <strong>Focusing the GWAS <em>Lens</em> on days to flower using
latent variable phenotypes derived from global multi-environment
trials</strong>. <em>The Plant Genome</em>. (<strong>2022</strong>)
e20269. doi.org/10.1002/tpg2.20269</a></p>
</blockquote>
<p>which is follow-up to:</p>
<blockquote>
<p><a href="https://doi.org/10.1002/ppp3.10158">Derek M. Wright, Sandesh
Neupane, Taryn Heidecker, Teketel A. Haile, Crystal Chan, Clarice J.
Coyne, Rebecca J. McGee, Sripada Udupa, Fatima Henkrar, Eleonora
Barilli, Diego Rubiales, Tania Gioia, Giuseppina Logozzo, Stefania
Marzario, Reena Mehra, Ashutosh Sarker, Rajeev Dhakal, Babul Anwar,
Debashish Sarker, Albert Vandenberg &amp; Kirstin E. Bett.
<strong>Understanding photothermal interactions can help expand
production range and increase genetic diversity of lentil (<em>Lens
culinaris</em> Medik.)</strong>. <em>Plants, People, Planet</em>.
(<strong>2021</strong>) 3(2): 171-181.
doi.org/10.1002/ppp3.10158</a></p>
</blockquote>
<hr />
<blockquote>
<ul>
<li><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology"
class="uri">https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology</a></li>
<li><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/README.pdf">View
as pdf</a></li>
<li><a
href="https://derekmichaelwright.github.io/AGILE_LDP_GWAS_Phenology/README.html">View
as HTML</a></li>
<li><a
href="https://derekmichaelwright.github.io/AGILE_LDP_GWAS_Phenology/GWAS_Phenology_Vignette.html">Source
Code Vignette (GWAS_Phenology_Vignette.html)</a></li>
</ul>
</blockquote>
<div id="contents" class="section level1">
<h1>Contents</h1>
<ul>
<li><a href="#figures">Figures</a></li>
<li><a href="#supplemental-tables">Supplemental Tables</a></li>
<li><a href="#supplemental-figures">Supplemental Figures</a></li>
<li><a href="#additional-figures">Additional Figures</a></li>
</ul>
</div>
<div id="agile-project" class="section level1">
<h1>AGILE Project</h1>
<p><img src="Additional/img_Agile.png" /></p>
<div style="page-break-after: always;"></div>
<div id="collaborators" class="section level2">
<h2>Collaborators</h2>
<ul>
<li>Department of Plant Sciences and Crop Development Centre, University
of Saskatchewan, Saskatoon, Saskatchewan, Canada</li>
<li>UTAZ</li>
</ul>
</div>
<div id="sponsors" class="section level2">
<h2>Sponsors</h2>
<ul>
<li>Saskatchewan Pulse Growers Association</li>
<li>Western Grains Research Foundation</li>
<li>GenomePrairie</li>
<li>GenomeCanada</li>
<li>Saskatchewan Ministry of Agriculture</li>
</ul>
<hr />
</div>
</div>
<div id="figures" class="section level1">
<h1>Figures</h1>
<div id="figure-1" class="section level2">
<h2>Figure 1</h2>
<p><img src="Figure_01.jpg" /></p>
<p><em>Figure 1: Genome-wide association results for days from sowing to
flower (DTF) in Sutherland, Canada 2017 (Su17), Jessore, Bangladesh 2017
(Ba17), Bardiya, Nepal 2017 (Ne17) and Metaponto, Italy 2017 (It17) for
a lentil diversity panel. (a) Histograms of DTF. (b) Manhattan and QQ
plots for GWAS results using MLM, MLMM, FarmCPU and Blink models.
Vertical lines represent specific base pair locations to facilitate
comparisons across traits.</em></p>
<hr />
<div style="page-break-after: always;"></div>
</div>
<div id="figure-2" class="section level2">
<h2>Figure 2</h2>
<blockquote>
<p><a
href="https://derekmichaelwright.github.io/AGILE_LDP_GWAS_Phenology/Additional/Figure_02_plotly.html">Additional/Figure_02_plotly.html</a></p>
</blockquote>
<p><img src="Figure_02.jpg" /></p>
<p><em>Figure 2: Summary of genome-wide association results using MLM,
MLMM, FarmCPU and Blink models on single environment and
multi-environment traits related to days from sowing to flower (DTF) in
a lentil diversity panel. Larger points represent a significant
association (-log10(p) &gt; 6.7) with a trait of interest under one of
the GWAS models, while smaller points represent a suggestive association
(-log10(p) &gt; 5.3). PC1, PC2, PC3, represent the first three principal
components of an analysis of 18 site-years of DTF data (Wright et al.,
2020) in Rosthern, Canada 2016 and 2017 (Ro16 and Ro17), Sutherland,
Canada 2016, 2017 and 2018 (Su16, Su17 and Su18), Central Ferry, USA
2018 (Us18), Bhopal, India 2016 and 2017 (In16 and In17), Jessore,
Bangladesh 2016 and 2017 (Ba16 and Ba17), Bardiya, Nepal 2016 and 2017
(Ne16 and Ne17), Marchouch, Morocco 2016 and 2017 (Mo16 and Mo17),
Cordoba, Spain (Sp16 and Sp17), Metaponto, Italy 2016 and 2017 (It16 and
It17). a, b and c are coefficients from a photothermal model (Wright et
al., 2020) and used to calculate the nominal base temperature (Tb),
nominal base photoperiod (Pc), thermal sum required for flowering (Tf)
and the photoperiod sum required for flowering (Pf). Colors are
representative of macroenvironments: Temperate (green), South Asian
(orange) Mediterranean (blue) and multi-environment traits (grey).
Vertical lines represent specific base pair locations to facilitate
comparisons across traits.</em></p>
<hr />
</div>
<div id="figure-3" class="section level2">
<h2>Figure 3</h2>
<p><img src="Figure_03.jpg" /></p>
<p><em>Figure 3: Genome-wide association results for the first three
principal components (PC1, PC2 and PC3) the b and c coefficients from
the photothermal model developed by Wright et al., (2020), and the
nominal base temperature (Tb) in Sutherland, Canada 2017 (Su17) for a
lentil diversity panel. (a) Histograms of each trait, scaled from 0 to
1. (b) Manhattan and QQ plots for GWAS results using MLM, MLMM, FarmCPU
and Blink models. Vertical lines represent specific base pair locations
to facilitate comparisons across traits.</em></p>
<hr />
</div>
<div id="figure-4" class="section level2">
<h2>Figure 4</h2>
<p><img src="Figure_04.jpg" /></p>
<p><em>Figure 4: Genome-wide association results for days from sowing to
flower (DTF) with and without covariates for a lentil diversity panel.
Manhattan and QQ plots for DTF in (a) Rosthern, Canada 2017 (Ro17) and
(b) Cordoba, Spain 2017 (Sp17), using MLM, MLMM, FarmCPU and Blink
models. The middle panel shows GWAS results without a covariate, while
the top and bottom panel show GWAS results using the c and b
(temperature and photoperiod) coefficients from the photothermal model
developed by Wright et al. 2020, respectively. Vertical lines represent
specific base pair locations to facilitate comparisons across
traits.</em></p>
<hr />
<div style="page-break-after: always;"></div>
</div>
<div id="figure-5" class="section level2">
<h2>Figure 5</h2>
<blockquote>
<p><a
href="https://derekmichaelwright.github.io/AGILE_LDP_GWAS_Phenology/Additional/Figure_05_a_plotly.html">Additional/Figure_05_a_plotly.html</a>
<a
href="https://derekmichaelwright.github.io/AGILE_LDP_GWAS_Phenology/Additional/Figure_05_b_plotly.html">Additional/Figure_05_b_plotly.html</a>
<a
href="https://derekmichaelwright.github.io/AGILE_LDP_GWAS_Phenology/Additional/Figure_05_c_plotly.html">Additional/Figure_05_c_plotly.html</a></p>
</blockquote>
<p><img src="Figure_05.jpg" /></p>
<p><em>Figure 5: Allelic effects of selected markers on days from sowing
to flower (DTF) across contrasting locations in a lentil diversity
panel. Sutherland, Canada 2017 (Su17), Jessore, Bangladesh 2017 (Ba17),
Bardiya, Nepal 2017 (Ne17) and Metaponto, Italy 2017 (It17). Colors are
based on a hierarchical clustering of principal components done by
Wright et al. (2020) using 18 site-years of DTF data across the three
major lentil growing macroenvironments. (a), (b) and (c) correspond to
SNPs at the vertical lines on Figures 1, 2, 3 &amp; 4.</em></p>
<hr />
</div>
</div>
<div id="supplemental-tables" class="section level1">
<h1>Supplemental Tables</h1>
<div id="supplemental-table-1" class="section level2">
<h2>Supplemental Table 1</h2>
<blockquote>
<p><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/blob/master/Supplemental_Table_01.csv">Supplemental_Table_01.csv</a></p>
</blockquote>
<p><em>Supplemental Table 1: GWAS results for SNPs significantly
associated with the traits of interest used in this study. Traits
include days from sowing to flower (DTF), the first three principal
components from a principal component analysis (PCA) of the DTF data
(PC1, PC2, PC3), the a, b and c coefficients from a photothermal model
(PTModel), the nominal base temperature (Tb), nominal base photoperiod
(Pc), thermal sum required for flowering (Tf) and the photoperiod sum
required for flowering (Pf). Rosthern, Canada 2016 and 2017 (Ro16 and
Ro17), Sutherland, Canada 2016, 2017 and 2018 (Su16, Su17 and Su18),
Central Ferry, USA 2018 (Us18), Bhopal, India 2016 and 2017 (In16 and
In17), Jessore, Bangladesh 2016 and 2017 (Ba16 and Ba17), Bardiya, Nepal
2016 and 2017 (Ne16 and Ne17), Marchouch, Morocco 2016 and 2017 (Mo16
and Mo17), Cordoba, Spain (Sp16 and Sp17), Metaponto, Italy 2016 and
2017 (It16 and It17). For further details see Wright et al. (2020).
Traits run with the b or c coefficients as a covariate are indicated
with the “-b” and “-c” suffix in the trait column.</em></p>
<hr />
</div>
<div id="supplemental-table-2" class="section level2">
<h2>Supplemental Table 2</h2>
<blockquote>
<p><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/blob/master/Supplemental_Table_02.xlsx">Supplemental_Table_02.xlsx</a></p>
</blockquote>
<p><em>Supplemental Table 2: List of genes in the regions associated
with flowering time in lentils chromosomes 2 and 5, and the syntenic
regions of chickpea (Cicer arietinum) and Medicago truncatula.</em></p>
<hr />
<div style="page-break-after: always;"></div>
</div>
</div>
<div id="supplemental-figures" class="section level1">
<h1>Supplemental Figures</h1>
<div id="supplemental-figure-1" class="section level2">
<h2>Supplemental Figure 1</h2>
<blockquote>
<p><a
href="https://derekmichaelwright.github.io/AGILE_LDP_GWAS_Phenology/Supplemental_Figure_01_plotly.html">Supplemental_Figure_01_plotly.html</a></p>
</blockquote>
<p><img src="Supplemental_Figure_01.jpg" /></p>
<p><em>Supplemental Figure 1: Summary of genome-wide association results
using MLM, MLMM, FarmCPU and Blink models on days from sowing to flower
(DTF) in a lentil diversity panel using no covariates or with the either
the b or c coefficient as a covariate. Larger points represent a
significant association (-log10(p) &gt; 6.7) with a trait of interest
under one of the GWAS models, while smaller points represent a
suggestive association (-log10(p) &gt; 5.3). Rosthern, Canada 2016 and
2017 (Ro16 and Ro17), Sutherland, Canada 2016, 2017 and 2018 (Su16, Su17
and Su18), Central Ferry, USA 2018 (Us18), Bhopal, India 2016 and 2017
(In16 and In17), Jessore, Bangladesh 2016 and 2017 (Ba16 and Ba17),
Bardiya, Nepal 2016 and 2017 (Ne16 and Ne17), Marchouch, Morocco 2016
and 2017 (Mo16 and Mo17), Cordoba, Spain (Sp16 and Sp17), Metaponto,
Italy 2016 and 2017 (it16 and It17). b and c are coefficients derived
from a photothermal model (Wright et al. (2020). Colors are
representative of macroenvironments: Temperate (green), South Asian
(orange) and Mediterranean (blue). Vertical lines represent specific
base pair locations to facilitate comparisons across traits.</em></p>
<hr />
</div>
<div id="supplemental-figure-2" class="section level2">
<h2>Supplemental Figure 2</h2>
<p><img src="Supplemental_Figure_02_edited_small.png" /></p>
<p><em>Supplemental Figure 2: Correlations along with the corresponding
correlation coefficients (R2) for the first three principal components
(PC1, PC2 &amp; PC3) of an analysis of 18 site-years of days from sowing
to flower (DTF) data (Wright et al., 2020), coefficients from a
photothermal model (a, b &amp; c), DTF, photoperiod sum required for
flowering (Pf), nominal base photoperiod (Pc), thermal sum required for
flowering (Tf) and nominal base temperature (Tb) in Sutherland, Canada
2017 (SU17), Jessore, Bangladesh 2017 (Ba17) and Metaponto, Italy 2017
(It17). Colors are based on a hierarchical clustering of the
PCAs.</em></p>
<hr />
</div>
<div id="supplemental-figure-3" class="section level2">
<h2>Supplemental Figure 3</h2>
<p><img src="Supplemental_Figure_03.jpg" style="width:70.0%" /></p>
<p><em>Supplemental Figure 3: Regional genome-wide association results
from 35 Mpb to 50 Mbp on chromosome 2 for selected traits with a lentil
diversity panel using MLM, MLMM, FarmCPU and Blink models. Traits
include days from sowing to flower (DTF) in Sutherland, Canada 2017
(Su17), with and without the c coefficient used as a covariate (CV=c),
DTF in Jessore, Bangladesh 2017 (Ba17), DTF in Bardiya, Nepal 2017
(Ne17), DTF in Metaponto, Italy 2017 (It17), the first principal
component (PC1) from a principal component analysis of an analysis of 18
site-years of DTF data, and the b coefficient derived from a
photothermal model done by Wright et al. (2020). The vertical line
represents a specific base pair location to facilitate comparisons
across traits.</em></p>
<hr />
</div>
<div id="supplemental-figure-4" class="section level2">
<h2>Supplemental Figure 4</h2>
<p><img src="Supplemental_Figure_04.jpg" style="width:70.0%" /></p>
<p><em>Supplemental Figure 4: Regional genome-wide association results
from 300 Mbp to 315 Mbp on chromosome 6 for selected traits with a
lentil diversity panel using MLM, MLMM, FarmCPU and Blink models. Traits
include days from sowing to flower (DTF) in Marchouch, Morocco 2016
(Mo16), thermal sum required for flowering (Tf) in Sutherland, Canada
2017 (Su17) and Metaponto, Italy 2017 (It17), nominal base temperature
(Tb) in Su17 and It17, and the third principal component (PC3) from a
principal component analysis of an analysis of 18 site-years of DTF data
(Wright et al., 2020). Vertical lines represent the locations of
selected flowering time genes within the associated QTL.</em></p>
<hr />
</div>
<div id="supplemental-figure-5" class="section level2">
<h2>Supplemental Figure 5</h2>
<p><img src="Supplemental_Figure_05.jpg" /></p>
<p><em>Supplemental Figure 5: Linkage disequilibrium decay across the 7
chromosomes in the lentil genome. (a) Linkage disequilibrium decay for
all marker combinations. (b) Linkage disequilibrium decay for marker
combinations within a 1 Mbp distance. For each chromosome, 1000 SNP were
randomly selected for pairwise LD calculations. Shaded lines represent
the moving average of 100 pair-wise marker comparisons. Solid line
represents a loess regression used to determine the value (vertical
dashed line) in which R2 reaches the 0.2 threshold (blue dashed line).
Red dashed lines represent the average R2 for each chromosome.</em></p>
<hr />
</div>
<div id="supplemental-figure-6" class="section level2">
<h2>Supplemental Figure 6</h2>
<p><img src="Supplemental_Figure_06.jpg" /></p>
<p><em>Supplemental Figure 6: Diagram of a 290 kb region of lentil
chromosome 2 (A) and a 215 kb region of lentil chromosome 5 (B)
containing the relevant SNPs Lcu.2RBY.Chr2p42543877,
Lcu.2RBY.Chr2p42556949 and Lcu.2RBY.Chr5p1063138 (highlighted in red).
The syntenic regions in chickpea and Medicago genomes are also shown for
comparison. The length of the relevant interval for each chromosome was
calculated according to the SNP position ± chromosome-specific linkage
disequilibrium decay. Numbers over or besides each gene correspond to
those shown in Supplemental Table 2.</em></p>
<hr />
<div style="page-break-after: always;"></div>
</div>
<div id="supplemental-figure-7" class="section level2">
<h2>Supplemental Figure 7</h2>
<blockquote>
<p><a
href="https://derekmichaelwright.github.io/AGILE_LDP_GWAS_Phenology/Supplemental_Figure_07_plotly.html">Supplemental_Figure_07_plotly.html</a></p>
</blockquote>
<p><img src="Supplemental_Figure_07.jpg" /></p>
<p><em>Supplemental Figure 7: Pair-wise plots of a principal component
analysis of genetic marker data from a lentil diversity panel. Colors
are based on a hierarchical clustering of principal components done by
Wright et al. (2020) using 18 site-years of days from sowing to flower
data across the three major lentil growing macroenvironments.</em></p>
<hr />
<div style="page-break-after: always;"></div>
</div>
</div>
<div id="additional-figures" class="section level1">
<h1>Additional Figures</h1>
<div id="additional-figure-1" class="section level2">
<h2>Additional Figure 1</h2>
<p><img src="Additional/Additional_Figure_01.jpg" /></p>
</div>
<div id="additional-figure-2" class="section level2">
<h2>Additional Figure 2</h2>
<p><img src="Additional/Additional_Figure_02.jpg" /></p>
</div>
<div id="additional-figure-3" class="section level2">
<h2>Additional Figure 3</h2>
<p><img src="Additional/Additional_Figure_03.jpg" /></p>
</div>
<div id="additional-figure-4" class="section level2">
<h2>Additional Figure 4</h2>
<p><img src="Additional/Additional_Figure_04.jpg" /></p>
</div>
<div id="additional-figure-5" class="section level2">
<h2>Additional Figure 5</h2>
<p><img src="Additional/Additional_Figure_05.jpg" /></p>
<hr />
<div style="page-break-after: always;"></div>
</div>
<div id="grouped-manhattan-plots" class="section level2">
<h2>Grouped Manhattan Plots</h2>
<blockquote>
<p><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/tree/master/Additional/Man_Grouped">Additional/Man_Grouped/</a></p>
</blockquote>
<p><img src="Additional/Man_Grouped/Man_PCA_abc.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_Temperate.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_b_Temperate.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_c_Temperate.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_Mediterranean.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_b_Mediterranean.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_c_Mediterranean.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_SouthAsia.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_b_SouthAsia.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_DTF_c_SouthAsia.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_Tf_Tb.jpg" /></p>
<p><img src="Additional/Man_Grouped/Man_Pf_Pc.jpg" /></p>
<hr />
<div style="page-break-after: always;"></div>
</div>
<div id="facetted-multi-modeled-manhattan-plots" class="section level2">
<h2>Facetted &amp; Multi-Modeled Manhattan Plots</h2>
<blockquote>
<p><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/tree/master/Additional/Man_Facet">Additional/Man_Facet/</a></p>
</blockquote>
<p><img src="Additional/Man_Facet/ManQQ_Su18_DTF.jpg" /></p>
<blockquote>
<p><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/tree/master/Additional/Man_Multi">Additional/Man_Multi/</a></p>
</blockquote>
<p><img src="Additional/Man_Multi/ManQQ_Su18_DTF.jpg" /></p>
<hr />
<div style="page-break-after: always;"></div>
</div>
<div id="marker-plots" class="section level2">
<h2>Marker Plots</h2>
<blockquote>
<p><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/Markers/Markers_Chr_1.pdf">Additional/Markers/Markers_Chr_1.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/Markers/Markers_Chr_2.pdf">Additional/Markers/Markers_Chr_2.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/Markers/Markers_Chr_3.pdf">Additional/Markers/Markers_Chr_3.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/Markers/Markers_Chr_4.pdf">Additional/Markers/Markers_Chr_4.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/Markers/Markers_Chr_5.pdf">Additional/Markers/Markers_Chr_5.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/Markers/Markers_Chr_6.pdf">Additional/Markers/Markers_Chr_6.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/Markers/Markers_Chr_7.pdf">Additional/Markers/Markers_Chr_7.pdf</a></p>
</blockquote>
<p><img
src="Additional/Markers/Markers_Lcu.2RBY.Chr2p42556949.jpg" /></p>
<p><img
src="Additional/Markers/Markers_Lcu.2RBY.Chr5p1069654.jpg" /></p>
<p><img
src="Additional/Markers/Markers_Lcu.2RBY.Chr6p2528817.jpg" /></p>
<p><img
src="Additional/Markers/Markers_Lcu.2RBY.Chr6p307256203.jpg" /></p>
<hr />
</div>
<div id="flowering-time-genes" class="section level2">
<h2>Flowering Time Genes</h2>
<blockquote>
<p><a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/LcFTa1.pdf">Additional/LcFTa1.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/LcFTb1.pdf">Additional/LcFTb1.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/LcELF3a.pdf">Additional/LcELF3a.pdf</a>
<a
href="https://github.com/derekmichaelwright/AGILE_LDP_GWAS_Phenology/raw/master/Additional/LcGI.pdf">Additional/LcGI.pdf</a></p>
</blockquote>
<hr />
<p>© Derek Michael Wright</p>
</div>
</div>
</div>

   
   
            
      

  <script>
    $(document).ready(function () {

			
 		
		$('#content img:not(.no-lightbox)').addClass("image-lb");
	$('#content').magnificPopup({
	    type:'image',
	    closeOnContentClick: false,
	    closeBtnInside: false,
	    delegate: '.image-lb',
	    gallery: {enabled: true },
	    image: {
	        verticalFit: true,
		titleSrc: 'alt'
	    }
 	});
 	    });
  </script>



    <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
	var script = document.createElement("script");
	script.type = "text/javascript";
	script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
	document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>
  
</body>
</html>
